Source/UI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -172,9 +172,17 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
         if shouldRemoveText {
             textValueOfCellType[type.index] = nil
         }
+        var areAllFieldsAreFulfilled = true
 
-        let areAllFieldsAreFulfilled = textValueOfCellType.values.count == style.cells.count && !hasErrorValue
-        editDelegate?.didFinishEditingBillingForm(successfully: areAllFieldsAreFulfilled)
+        for cell in style.cells where cell.style?.isMandatory ?? false {
+            let isEmpty = textValueOfCellType[cell.index]?.isEmpty
+            guard isEmpty == false else {
+                areAllFieldsAreFulfilled = false
+                break
+            }
+        }
+
+        editDelegate?.didFinishEditingBillingForm(successfully: areAllFieldsAreFulfilled && !hasErrorValue)
```

Comment:
I think we should be able to modify this syntax using `first(where: ` (not valid) and checking if such a value exists. I'll try to suggest syntax (lack of time now), but a unit tests for this would be very useful.

Reply:
IMHO, `first(where:` has differnt usage and not suitable for this case because it's searching and return first value of type `CellTextFieldStyle` where some condition. in this situation there's only check on mandatory value then another check if it's empty then change `areAllFieldsAreFulfilled` then break.

Reply:
Exactly why you don't need `areAllFieldsAreFulfilled`. Ignore that and rethink what you're looking for.

You're looking to see **if any field is not fulfilled**. 
You can tell that the first time you meet an item that is not fulfilled (this is the `first`). 
If such an early return is not met, you know you've successfully validated all fields are fulfilled!

In code that means something in the lines of:
```swift
let array = ["a", "b"]
let hasC = array.first(where: { $0 == "c" }) != nil
```

your logic appears slightly more complex, which is why I can only give you a generic example. If it still doesn't make sense I'll switch to your branch and find the actual code you need.

Reply:
Got the code, should be
```Swift
// Most on point
let areAllFieldFulfilled = style.cells.first(where: {
  guard let style = $0.style
            style.isMandatory else {
               return false
  }
  return textValueOfCellType[$0.index]?.isEmpty == true
}) == nil


// Breaking down the logic
let incompleteField = style.cells.first(where: {
  guard let style = $0.style
            style.isMandatory else {
               return false
  }
  return textValueOfCellType[$0.index]?.isEmpty == true
})
let areAllFieldsFulfilled = incompleteField == nil
```